src/EnergyPlus/HeatingCoils.cc

```diff
@@ -221,28 +221,36 @@ namespace HeatingCoils {
         InitHeatingCoil(state, CoilNum, FirstHVACIteration, QCoilRequired); // Initialize all HeatingCoil related parameters
 
         // Calculate the Correct HeatingCoil Model with the current CoilNum
-        if (state.dataHeatingCoils->HeatingCoil(CoilNum).HCoilType_Num == Coil_HeatingElectric) {
+        switch (state.dataHeatingCoils->HeatingCoil(CoilNum).HCoilType_Num) {
+        case Coil_HeatingElectric: {
             CalcElectricHeatingCoil(state, CoilNum, QCoilRequired, QCoilActual2, OpMode, PartLoadFrac);
-        } else if (state.dataHeatingCoils->HeatingCoil(CoilNum).HCoilType_Num == Coil_HeatingElectric_MultiStage) {
+        } break;
+        case Coil_HeatingElectric_MultiStage: {
             CalcMultiStageElectricHeatingCoil(state,
                                               CoilNum,
                                               SpeedRatio,
                                               PartLoadRatio,
                                               StageNum,
-                                              OpMode); // Autodesk:OPTIONAL SpeedRatio, PartLoadRatio, StageNum used without PRESENT check
-        } else if (state.dataHeatingCoils->HeatingCoil(CoilNum).HCoilType_Num == Coil_HeatingGasOrOtherFuel) {
+                                              OpMode,
+                                              QCoilActual2); // Autodesk:OPTIONAL SpeedRatio, PartLoadRatio, StageNum used without PRESENT check
```

Comment:
Regarding diffs, the other change I see is you are now passing &QCoilActual so I am wondering if this fixed something now that the Calc function has more information.